frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditCheckbox.tsx

```diff
@@ -0,0 +1,279 @@
+    watch,
+    control,
+    clearErrors,
+  } = useEditFieldForm<EditCheckboxInputs, CheckboxFieldBase>({
+    ...props,
+    transform: {
+      input: transformCheckboxFieldToEditForm,
+      output: transformCheckboxEditFormToField,
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const watchedInputs = watch()
+
+  const customMinValidationOptions: RegisterOptions = useMemo(
+    () => ({
+      required: {
+        value:
+          watchedInputs.validateByValue &&
+          !watchedInputs.ValidationOptions.customMax,
+        message: 'Please enter selection limits',
+      },
+      min: {
+        value: 1,
+        message: 'Cannot be less than 1',
+      },
+      validate: {
+        minLargerThanMax: (val) => {
+          return (
+            !val ||
+            !watchedInputs.validateByValue ||
+            !watchedInputs.ValidationOptions.customMax ||
+            Number(val) <= Number(watchedInputs.ValidationOptions.customMax) ||
+            'Minimum cannot be larger than maximum'
+          )
+        },
+        max: (val) => {
+          let numOptions = SPLIT_TEXTAREA_TRANSFORM.output(
+            watchedInputs.fieldOptions,
+          ).length
+          if (watchedInputs.othersRadioButton) {
+            numOptions += 1
+          }
+          return (
+            !val || val <= numOptions || 'Cannot be more than number of options'
+          )
+        },
+      },
+    }),
+    [watchedInputs],
+  )
+
+  const customMaxValidationOptions: RegisterOptions = useMemo(
+    () => ({
+      required: {
+        value:
+          watchedInputs.validateByValue &&
+          !watchedInputs.ValidationOptions.customMin,
+        message: 'Please enter selection limits',
+      },
+      min: {
+        value: 1,
+        message: 'Cannot be less than 1',
+      },
+      validate: {
+        maxLargerThanMin: (val) => {
+          return (
+            !val ||
+            !watchedInputs.validateByValue ||
+            !watchedInputs.ValidationOptions.customMin ||
+            Number(val) >= Number(watchedInputs.ValidationOptions.customMin) ||
+            'Maximum cannot be less than minimum'
+          )
+        },
+        max: (val) => {
+          if (!watchedInputs.validateByValue) return true
+          let numOptions = SPLIT_TEXTAREA_TRANSFORM.output(
+            watchedInputs.fieldOptions,
+          ).length
+          if (watchedInputs.othersRadioButton) {
+            numOptions += 1
+          }
+          return (
+            !val || val <= numOptions || 'Cannot be more than number of options'
+          )
+        },
+      },
+    }),
+    [watchedInputs],
+  )
+
+  // Effect to clear validation option errors when selection limit is toggled off.
+  useEffect(() => {
+    if (!watchedInputs.validateByValue) {
+      clearErrors('ValidationOptions')
+    }
+  }, [clearErrors, watchedInputs.validateByValue])
```

Comment:
nit: maybe we can use the [`deps`](https://react-hook-form.com/api/useform/register#:~:text=Validation%20will%20be%20triggered%20for%20the%20dependent%20inputs) prop in `register` instead of a useEffect to achieve the same result

Reply:
tried this, unfortunately it didn't work, I think because the input value remains the same. even if the validation is re-triggered. e.g. if I put 0 as the min selection limit (invalid because less than 1) then I deselect the "Selection limits" toggle, the input will still be invalid.

I also tried using `Controller` and a custom `onChange` to unset the `customMin` and `customMax` when Selection Limits is toggled off, but ran into the same bug where setting values to `undefined` results in form getting set back to previous state. then tried setting to empty string instead of `undefined`, but ran into type errors (since these are supposed to be numbers) and also the errors still didn't get cleared even after running `trigger`, possibly because the `rules` only get updated on the following render.

think `useEffect` might be the best solution for now! wdyt?

Reply:
Yea, can continue using useEffect then! But note that it may be hiding some bugs in that case haha